process.c

```diff
@@ -619,18 +619,30 @@ proc_s_last_status(VALUE mod)
     return rb_last_status_get();
 }
 
+static VALUE
+rb_process_status_initialize(VALUE self, VALUE pid, VALUE status, VALUE error)
+{
+    struct rb_process_status *data = RTYPEDDATA_DATA(self);
+
+    data->pid = NUM2PIDT(pid);
+    data->status = NUM2INT(status);
+    data->error = NUM2INT(error);
+
+    return self;
+}
+
 VALUE
 rb_process_status_new(rb_pid_t pid, int status, int error)
 {
-    VALUE last_status = rb_process_status_allocate(rb_cProcessStatus);
+    VALUE self = rb_process_status_allocate(rb_cProcessStatus);
 
-    struct rb_process_status *data = RTYPEDDATA_DATA(last_status);
+    struct rb_process_status *data = RTYPEDDATA_DATA(self);
     data->pid = pid;
     data->status = status;
     data->error = error;
 
-    rb_obj_freeze(last_status);
-    return last_status;
+    rb_obj_freeze(self);
+    return self;
```

Comment:
Seems an unrelated change.

Reply:
Make it consistent with `initialize`. `last_status` doesn't mean anything.

Reply:
It is not "self" here.

To make consistent, how about extracting a function.

```C
static VALUE
process_status_init(VALUE self, rb_pid_t pid, int status, int error)
{
    struct rb_process_status *data = RTYPEDDATA_DATA(self);
    data->pid = pid;
    data->status = status;
    data->error = error;
    rb_obj_freeze(self);
    return self;
}

VALUE
rb_process_status_new(rb_pid_t pid, int status, int error)
{
    VALUE obj = rb_process_status_allocate(rb_cProcessStatus);
    return process_status_init(obj, pid, status, error);
}

static VALUE
rb_process_status_initialize(VALUE self, VALUE vpid, VALUE vstatus, VALUE verror)
{
    rb_pid_t pid = NUM2PIDT(vpid);
    int status = NUM2INT(vstatus);
    int error = NUM2INT(verror);
    return process_status_init(self, pid, status, error);
}
```

Reply:
I agree, done.